pkg/cli/copy.go

```diff
@@ -56,17 +56,23 @@ func (a *ImageCopy) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	if !a.AllTags {
+		if _, err := c.ImageGet(cmd.Context(), args[0]); err == nil {
+			return a.copyLocalToRemote(cmd, c, args, creds)
+		}
+	}
+
 	source, err := name.ParseReference(args[0])
 	if err != nil {
 		return err
 	}
 
-	dest, err := name.ParseReference(args[1])
+	sourceAuth, _, err := creds.Get(cmd.Context(), source.Context().RegistryStr())
```

Comment:
Is there a reason why these are being switched?

Reply:
Nope, the reordering was unintentional as I moved stuff around. I can switch the order back if you think it makes more sense the other way.

Reply:
No, it's totally fine. I was more or less wondering if this was important and needed a comment, too.